Realm/RLMRealm.mm

```diff
@@ -944,16 +950,24 @@ - (RLMObject *)createObject:(NSString *)className withValue:(id)value {
 }
 
 - (BOOL)writeCopyToURL:(NSURL *)fileURL encryptionKey:(NSData *)key error:(NSError **)error {
+    return [self writeCopyToURL:fileURL encryptionKey:key enableSync:NO error:error];
+}
+
+- (BOOL)writeCopyToURL:(NSURL *)fileURL encryptionKey:(NSData *)key enableSync:(BOOL)enableSync error:(NSError **)error {
     key = RLMRealmValidatedEncryptionKey(key);
     NSString *path = fileURL.path;
 
     try {
-        _realm->verify_thread();
-        try {
-            _realm->read_group().write(path.UTF8String, static_cast<const char *>(key.bytes));
-        }
-        catch (...) {
-            _impl::translate_file_exception(path.UTF8String);
+        if (enableSync) {
+            _realm->write_copy(path.UTF8String, {static_cast<const char *>(key.bytes), key.length});
```

Comment:
This call needs to be inside the try/catch which translates file exceptions. Currently if any errors occur this'll result in an c++ exception escaping, and we need to translate it to an obj-c exception.